sympy/functions/special/delta_functions.py

```diff
@@ -159,6 +159,36 @@ def is_simple(self, x):
             return p.degree() == 1
         return False
 
+    def _eval_rewrite_as_Piecewise(self, *args):
+        """Represents DiracDelta in a Piecewise form
+
+           Examples
+           ========
+
+           >>> from sympy import DiracDelta, Piecewise, Symbol, oo
+           >>> x = Symbol('x', real=True)
+
+           >>> DiracDelta(x).rewrite(Piecewise)
+           Piecewise((oo, Eq(x, 0)), (0, True))
+
+           >>> DiracDelta(x - 5).rewrite(Piecewise)
+           Piecewise((oo, Eq(x - 5, 0)), (0, True))
+
+           >>> DiracDelta(x - 5, 4).rewrite(Piecewise)
+           Piecewise((oo, Eq(x - 5, 0)), (0, True))
+
+           >>> y = Symbol('y')
+
+           >>> DiracDelta(y).rewrite(Piecewise)
+           DiracDelta(y)
```

Comment:
@asmeurer Yeah, there is no hard and fast rule for setting this assumption. But I saw this assumption being implemented in two places.

- [Heaviside](https://github.com/sympy/sympy/blob/master/sympy/functions/special/delta_functions.py#L239)
- [sign](https://github.com/sympy/sympy/blob/master/sympy/functions/elementary/complexes.py#L359) 

We can remove this assumption.  I also agree to this:

> If someone writes DiracDelta(y) that's enough to implicitly assume that y is real.

Ok then, I am removing the check for real.